File: Data Structures & Algorithms/validate-parentheses/submission-0.py
class Solution:
    def isValid(self, s: str) -> bool:
        st = []
        top = -1 

        for ch in s:
            if ch in ['[', '{', '(']:
                st.append(ch)
                top += 1
            else:
                if top == -1:  
                    return False
                if (ch == ']' and st[top] != '[') or \
                   (ch == '}' and st[top] != '{') or \
                   (ch == ')' and st[top] != '('):
                    return False
                st.pop()
                top -= 1

        return top == -1

File: Data Structures & Algorithms/pow-x-n/submission-0.java
class Solution {
    public double myPow(double x, int n) {
        if(n<0) return 1/myPow(x,-n);
        if(n==0) return 1;
        double h=myPow(x,n/2);
        if(n%2==0) return h*h;
        else return h*h*x;
    }
}
